Ds3/Calls/PutObjectRequest.cs

```diff
@@ -74,7 +74,14 @@ public PutObjectRequest WithMetadata(IDictionary<string, string> metadata)
             }
             foreach (var keyValuePair in metadata)
             {
-                this.Headers.Add(HttpHeaders.AwsMetadataPrefix + keyValuePair.Key, keyValuePair.Value);
+                if (string.IsNullOrEmpty(keyValuePair.Value))
+                {
+                    System.Diagnostics.Trace.TraceWarning("Key has not been added to metadata because value was null or empty: " + keyValuePair.Key);
```

Comment:
We use the TraceSwitch to log messages, you can look for a referance in the Network.cs